mlrun/runtimes/nuclio/application/application.py

```diff
@@ -290,6 +316,77 @@ def with_source_archive(
             target_dir=target_dir,
         )
 
+    @classmethod
+    def get_filename_and_handler(cls) -> (str, str):
+        reverse_proxy_file_path = pathlib.Path(__file__).parent / "reverse_proxy.go"
+        return str(reverse_proxy_file_path), "Handler"
+
+    def create_api_gateway(
+        self,
+        path: str = None,
+        ports: list[int] = None,
+        skip_access_key_auth: bool = False,
+    ):
+        api_gateway = APIGateway(
+            APIGatewayMetadata(
+                name=self.metadata.name,
+                namespace=self.metadata.namespace,
+                labels=self.metadata.labels,
+                annotations=self.metadata.annotations,
+            ),
+            APIGatewaySpec(
+                functions=[self],
+                project=self.metadata.project,
+                path=path,
+                ports=mlrun.utils.helpers.as_list(ports) if ports else None,
+            ),
+        )
+
+        if not skip_access_key_auth:
+            api_gateway.with_access_key_auth()
+
+        db = self._get_db()
+        api_gateway_scheme = db.store_api_gateway(
+            api_gateway.to_scheme(), self.metadata.project
+        )
+        self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
+        self.status.api_gateway.wait_for_readiness()
+
+    def invoke(
+        self,
+        path: str,
+        body: typing.Union[str, bytes, dict] = None,
+        method: str = None,
+        headers: dict = None,
+        dashboard: str = "",
+        force_external_address: bool = False,
+        auth_info: AuthInfo = None,
+        mock: bool = None,
+        **http_client_kwargs,
+    ):
+        # If the API Gateway is not ready or not set, try to invoke the function directly (without the API Gateway)
+        if not self.status.api_gateway:
+            super().invoke(
+                path,
+                body,
+                method,
+                headers,
+                dashboard,
+                force_external_address,
+                auth_info,
+                mock,
+                **http_client_kwargs,
+            )
+
+        auth = (auth_info.username, auth_info.password) if auth_info else None
+
+        return self.status.api_gateway.invoke(
+            method=method or "POST",
+            headers=headers,
+            auth=auth,
+            path=path,
+        )
```

Comment:
The way we work here with `auth` looks a bit confusing. Due to introduction of `accessKey` auth, it makes sense from user perspective that `auth` param represents any authentication if it's needed. However, in this implementation `auth` represents only `basicAuth` and has nothing to do with `accessKey`. I think we should either rename `auth` to `basic_auth` or smth like this or rework the logic